SIPS/sip-3.md

```diff
@@ -93,63 +98,71 @@ interface OnTransactionArgs {
 }
 ```
 
-**Transaction** - The transaction object is specifically not defined in this SIP because the transaction object can look different across various chains and it is not our intention to define an interface for every chain. Instead, the onus is on the Snap developer to be cognizant of the shape of the transaction object. However, that being said, the _default_ transaction object in the MetaMask extension is of the following interface:
+`transaction` - The transaction object is intentionally not defined in this SIP because different chains may specify different transaction formats.
+It is beyond the scope of the SIP standards to define interfaces for every chain.
+Instead, it is the Snap developer's responsibility to be cognizant of the shape of transaction objects for relevant chains.
+Nevertheless, you can refer to [Appendix I](#appendix-i-ethereum-transaction-objects) for the interfaces of the Ethereum transaction objects available in MetaMask at the time of this SIP's creation.
+
+`chainId` - This is a [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md) `chainId` string.
+The snap is expected to parse and utilize this string as needed.
 
-_[NON EIP-1559]_
+The interface for the return value of an `onTransaction` export is:
 
 ```typescript
-interface LegacyTransactionObject {
+interface OnTransactionReturn {
+  insights: Record<string, Json>;
+}
+```
+
+### MetaMask Integration
+
+The `insights` object returned by the snap will be displayed alongside the confirmation for the `transaction` that `onTransaction` was called with.
+Keys and values will be displayed in the order received, with each key rendered as a title and each value rendered as follows:
+
+- If the value is an array or an object, it will be rendered as text after being converted to a string.
+- If the value is neither an array nor an object, it will be rendered directly as text.
+
+## Appendix I: Ethereum Transaction Objects
```

Comment:
We want to divorce the transaction objects from the specification itself. Arguably, we should find some external resource to link to, but an appendix is fine for now.

Reply:
Makes sense.